aws-s3/entrypoint.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/bash
+
+function usage_docs {
+  echo ""
+  echo "- uses: SonarSource/gh-action_release/aws-s3@v2"
+  echo "  with:"
+  echo "    command: cp"
+  echo "    source: ./local_file.txt"
+  echo "    destination: s3://yourbucket/folder/local_file.txt"
+  echo "    aws_access_key_id: \${{ secret.AWS_ACCESS_KEY_ID }}"
+  echo "    aws_secret_access_key: \${{ secret.AWS_SECRET_ACCESS_KEY }}"
+  echo "    aws_session_token: \${{ secret.AWS_SESSION_TOKEN }}"
+  echo ""
+}
+
+function get_configuration_settings {
+  if [ -z "$INPUT_AWS_ACCESS_KEY_ID" ]
+  then
+    echo "AWS Access Key Id was not provided. Using configuration from previous step."
+  else
+    aws configure set aws_access_key_id "$INPUT_AWS_ACCESS_KEY_ID"
+  fi
+
+  if [ -z "$INPUT_AWS_SECRET_ACCESS_KEY" ]
+  then
+    echo "AWS Secret Access Key was not provided. Using configuration from previous step."
+  else
+    aws configure set aws_secret_access_key "$INPUT_AWS_SECRET_ACCESS_KEY"
+  fi
+
+  if [ -z "$INPUT_AWS_SESSION_TOKEN" ]
+  then
+    echo "AWS Session Token was not provided. Using configuration from previous step."
+  else
+    aws configure set aws_session_token "$INPUT_AWS_SESSION_TOKEN"
+  fi
+
+  if [ -z "$INPUT_METADATA_SERVICE_TIMEOUT" ]
+  then
+    echo "Metadata service timeout was not provided. Using configuration from previous step."
+  else
+    aws configure set metadata_service_timeout "$INPUT_METADATA_SERVICE_TIMEOUT"
+  fi
+
+  if [ -z "$INPUT_AWS_REGION" ]
+  then
+    echo "AWS region not provided. Using configuration from previous step."
+  else
+    aws configure set region "$INPUT_AWS_REGION"
+  fi
+}
+
+function get_command {
+  VALID_COMMANDS=("sync" "mb" "rb" "ls" "cp" "mv" "rm")
+  COMMAND="cp"
+  if [ -z "$INPUT_COMMAND" ]
+  then
+    echo "Command not set using cp"
+  elif [[ ! ${VALID_COMMANDS[*]} =~ "$INPUT_COMMAND" ]]
+  then
+    echo ""
+    echo "Invalid command provided :: [$INPUT_COMMAND]"
+    usage_docs
+    exit 1
+  else
+    echo "Using provided command"
+    COMMAND=$INPUT_COMMAND
+  fi
+}
+
+function validate_source_and_destination {
+  if [ "$COMMAND" == "cp" ] || [ "$COMMAND" == "mv" ] || [ "$COMMAND" == "sync" ]
+  then
+    # Require source and target
+    if [ -z "$INPUT_SOURCE" ] && [ "$INPUT_DESTINATION" ]
+    then
+      echo ""
+      echo "Error: Requires source and destination."
+      usage_docs
+      exit 1
+    fi
+
+    # Verify at least one source or target have s3:// as prefix
+    # if [[] || []]
+    if [[ ! "$INPUT_SOURCE" =~ ^s3:// ]] && [[ ! "$INPUT_DESTINATION" =~ ^s3:// ]]
+    then
+      echo ""
+      echo "Error: Source or target must have s3:// as prefix."
+      usage_docs
+      exit 1
+    fi
+  else
+    # Require source
+    if [ -z "$INPUT_SOURCE" ]
+    then
+      echo "Error: Requires source."
+      usage_docs
+      exit 1
+    fi
+
+    # Verify at source has s3:// as prefix
+    if [[ ! $INPUT_SOURCE =~ ^s3:// ]]
```

Comment:
vs https://github.com/SonarSource/gh-action_release/pull/117/files#diff-6cd290a0b2f8ecca5dcb4428bc6549b5d757317adef53f382cbdd4002bac0076R85 ?

"Source or target" or "source must" ?

Reply:
I will keep it as it is: If you pass only a source as parameter, the error message should not talk about target only source.